Lib/test/test_concurrent_futures/executor.py

```diff
@@ -71,6 +73,54 @@ def test_map_timeout(self):
 
         self.assertEqual([None, None], results)
 
+    def test_map_buffersize(self):
+        integers = range(8)
+
+        with self.assertRaisesRegex(
+            ValueError,
+            "buffersize must be None or > 0",
+            msg="`Executor.map` should raise if `buffersize` is not positive.",
+        ):
+            self.executor.map(str, integers, buffersize=0)
+
+        for buffersize in (1, 2, len(integers), len(integers) * 2):
+            self.assertEqual(
```

Comment:
Personally, I liked the previous splitting because each test can be run independently of the others. If you want to refactor your test, you can create a method instead (but you don't need to have one generic method for all tests).

If you insist on a single test here is some advice: 

- Instead of using 'msg', use `subTest` and indicate which buffersize is failing as well, e.g.:

```py
def do(cb, it, bs):
	return self.executor.map(cb, it, buffersize=bs)

with self.subTest("description of the test", buffersize=buffersize):
	res = do(str, integers)
	expect = list(map(...))
	self.assertListEqual(list(res), expect)
```

- Limit yourself to 80-chars per line and use intermediate variables so that we know which line is failing. For instance, for `next(self.executor.map(str, itertools.count(), buffersize=buffersize))`, we can write it as 

```py
res = self.executor.map(...)
peek = next(res, None)
self.assertEqual(peek, "0")
```

instead of computing `next(map(str, integers))`.  Similarly, for `self.executor.map(str, [], buffersize=buffersize))`, you can do 

```py
res = self.executor.map(str, [], buffersize=buffersize)
self.assertIsNone(next(res, None))
```

Reply:
Thank you for these guidelines @picnixz 🙏🏻 !

I have split them back into multiple methods (which I prefer too) and applied your guidelines regarding intermediate variables, chars/line, and subtests within loops.